wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+            wx.Frame.__init__(self, parent, id, title, pos, size, style, name)
+            ## self.DefineNativeCheckBoxBitmaps()
+            ## self.checkbox_bitmaps = self.GetNativeCheckBoxBitmaps()
+            cb1 = GenCheckBox(self, label="PurePython Checkbox1", pos=(10, 10))
+            cb2 = GenCheckBox(self, label="PurePython Checkbox2", pos=(10, 50))
+            cb1.Bind(wx.EVT_CHECKBOX, self.OnCheckBox)
+            cb2.Bind(wx.EVT_CHECKBOX, self.OnCheckBox)
+            cb2.SetForegroundColour(wx.GREEN)
+            cb2.SetBackgroundColour(wx.BLACK)
+            sizer = wx.BoxSizer()
+            sizer.Add(cb1, 0, wx.ALL, 5)
+            sizer.Add(cb2, 0, wx.ALL, 5)
+            self.SetSizer(sizer)
+
+        def OnCheckBox(self, event):
+            evtObj = event.GetEventObject()
+            print(evtObj.GetLabel(), evtObj.IsChecked())
+
+    frame = MyFrame(None, wx.ID_ANY, "Test Pure-Py Checkbox")
+    frame.Show()
+    app.MainLoop()
+
+"""
+
+# Imports.---------------------------------------------------------------------
+
+# -wxPython Imports.
+import wx
+
+
+class GenCheckBox(wx.Control):
+    """
+    A generic class that replicates some of the functionalities of :class:`wx.Checkbox`,
+    while being completely owner-drawn with a nice check bitmaps.
+    """
+
+    def __init__(self, parent, id=wx.ID_ANY, label="", pos=wx.DefaultPosition,
+                 size=wx.DefaultSize, style=wx.NO_BORDER, validator=wx.DefaultValidator,
+                 name="GenCheckBox"):
+        """
+        Default class constructor.
+
+        :param `parent`: Pointer to a parent window. Must not be ``None``.
+        :type `parent`: `wx.Window`
+        :param `id`: Window identifier. ``wx.ID_ANY`` indicates a default value.
+        :type `id`: int
+        :param `label`: Text to be displayed next to the checkbox.
+        :type `label`: str
+        :param `pos`: Window position. The value ``wx.DefaultPosition`` indicates
+         a default position, chosen by either the windowing system or wxWidgets, depending on platform.
+        :type `pos`: `wx.Point`
+        :param `size`: Window size. The value ``wx.DefaultSize`` indicates a default size,
+         chosen by either the windowing system or wxWidgets, depending on platform.
+        :type `size`: `wx.Size`
+        :param `style`: Window style. Not used in this widget, GenCheckBox has only 2 state.
+        :type `style`: long
+        :param `validator`: Window validator.
+        :type `validator`: `wx.Validator`
+        :param `name`: Window name.
+        :type `name`: str
+        """
+
+        # Ok, let's see why we have used wx.PyControl instead of wx.Control.
+        # Basically, wx.PyControl is just like its wxWidgets counterparts
+        # except that it allows some of the more common C++ virtual method
+        # to be overridden in Python derived class. For GenCheckBox, we
+        # basically need to override DoGetBestSize and AcceptsFocusFromKeyboard.
```

Comment:
Using wx.PyControl is no longer needed, so you can remove this comment about why you're using it.

Reply:
removed